### PR TITLE
Using GAPIC datastore object (and an HTTP equivalent) for allocate_ids.

### PR DESCRIPTION
**Pull request by dhermes**
Towards #2746.

### CODE REVIEW

File: datastore/google/cloud/datastore/_http.py [hunk 1]
```diff
@@ -451,25 +434,6 @@ def rollback(self, project, transaction_id):
...
-
-        :rtype: :class:`.datastore_pb2.AllocateIdsResponse`
-        :returns: The protobuf response from an allocate IDs request.
-        """
-        request = _datastore_pb2.AllocateIdsRequest()
-        _add_keys_to_request(request.keys, key_pbs)
-        return self._datastore_api.allocate_ids(project, request)
-
 
 class HTTPDatastoreAPI(object):
```
Comment (daspecster) [inline datastore/google/cloud/datastore/_http.py]: ` class _DatastoreAPIOverHttp` vs `class HTTPDatastoreAPI`??

File: datastore/google/cloud/datastore/_http.py [hunk 2]
```diff
@@ -250,23 +250,6 @@ def rollback(self, project, request_pb):
                     self.connection.api_base_url,
                     request_pb, _datastore_pb2.RollbackResponse)
 
-    def allocate_ids(self, project, request_pb):
```
Comment (daspecster) [inline datastore/google/cloud/datastore/_http.py]: No other calls directly to this or the other one?

**Comment by daspecster** [comment]
Looks pretty good to me.

Slightly off topic, what's the setup for the system tests for datastore?

File: datastore/google/cloud/datastore/_http.py (hunk 1 above)
Comment (dhermes) [inline datastore/google/cloud/datastore/_http.py]: `_DatastoreAPIOverHttp` and `Connection` are going to the junkyard once this process is finished.

**Comment by dhermes** [comment]
> what's the setup for the system tests for datastore?

Can you be more specific? Typically it's `tox -e system-tests -- datastore`, but there is an [extra instruction][1] for datastore for populating datastore indexes.

[1]: https://github.com/GoogleCloudPlatform/google-cloud-python/blob/ada0d2e68050fa7599f0521c68b9a69996471819/CONTRIBUTING.rst#running-system-tests

**Comment by daspecster** [comment]
@dhermes sorry yes, that section in CONTRIBUTING.rst was what I was looking for.

File: datastore/google/cloud/datastore/_http.py (hunk 1 above)
Comment (daspecster) [inline datastore/google/cloud/datastore/_http.py]: ok cool

**Pull request merged**